lib/mongoid/slug.rb

```diff
@@ -90,8 +91,18 @@ def slug(*fields, &block)
 
         # Set indexes
         if slug_index && !embedded?
-          Mongoid::Slug::IndexBuilder.build_indexes(self, slug_scope_key, slug_by_model_type,
-                                                    options[:localize])
+          # Check if scope is an array and handle accordingly.
+          if slug_scope.is_a?(Array)
```

Comment:
Probably simpler to do `Array(slug_scope).each do |...|`, avoids code duplication.